receiver/prometheus/metrics_receiver.go

```diff
@@ -15,36 +15,90 @@
 package prometheus
 
 import (
+	"context"
 	"errors"
+	"sync"
 	"time"
 
+	agentmetricspb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1"
+	"github.com/census-instrumentation/opencensus-service/data"
 	"github.com/census-instrumentation/opencensus-service/receiver"
+	"github.com/orijtech/promreceiver"
```

Comment:
Do we have a plan to move the receiver to this repo?

Reply:
I wouldn't advise doing that right now, but it can be done. The Prometheus receiver code is about ~1K lines long because we have to translate them on notification from Prometheus listeners, from their labels to our metrics.

That repo is licensed for OpenCensus Authors and preferably, I'd rather put it under census-ecosystem so that it can be worked on without blocking work on this repo.